api/v1/services/UserService.js

```diff
@@ -0,0 +1,93 @@
+var Checkit = require('checkit');
+var Promise = require('bluebird');
+var _ = require('lodash');
+
+var User = require('../models/User');
+var errors = require('../errors');
+var utils = require('../utils');
+
+function _findUserByEmail (email) {
+	if (email) {
+		email = email.toLowerCase();
+		return User.query().where('Email', email).select();
+	}
+
+	return Promise.resolve([]);
+}
+
+function _findUserById (id) {
+	if (id) {
+		return User.query().where('ID', id).select();
+	}
+
+	return Promise.resolve([]);
+}
+
+module.exports.createUser = function (email, password, role) {
+	email = email.toLowerCase();
+	password = (password) ? password : utils.crypto.generatePassword();
```

Comment:
Yeah I suppose we could do that in the `HackerCreationRequest` (and whatever creation request comes when multi-role registration is implemented).